### DISCUSSION TITLE
MAUi Opens Browser instead of opening url in windows app

### DISCUSSION THREAD

**Discussion by OlsonFernandes** [dotnet/maui#5527] [Q&A]
Navigation manager load external url 
Load it into windows emlator

**Comment by Shellishack** [comment]
Hi there,

I had a similar problem in my application. I managed to solve it via creating **Navigating** event handler. You can catch target url using **WebNavigatingEventArgs** argument, then cancel the navigation in your app and open target url in a browser.

First, create an event handler in your WebView element.
`<WebView Source="http://localhost:8200" Navigating="WebView_Navigating"/>`

Then catch the target url in your event handler. 
```
private void WebView_Navigating(object sender, WebNavigatingEventArgs e)
{
    UrlWebViewSource source = e.Source as UrlWebViewSource;
    if (!source.Url.StartsWith("http://localhost:8200/"))
    {
        e.Cancel = true;
        Browser.Default.OpenAsync(source.Url, BrowserLaunchMode.SystemPreferred);
    }
}
```
Here the URLs that do not start with http://localhost:8200 are sent to a browser. You can optionally customize your own filter if needed.
 
I am running this on a Windows Machine in this example. If you want to do this for other platforms, you can find more about cross-platform browser APIs here: [https://docs.microsoft.com/en-us/dotnet/maui/platform-integration/appmodel/open-browser](url)
